tests/fixture/database/test_database.py

```diff
@@ -101,3 +101,13 @@ def test_bad_actions(postgres):
     assert "create_fixture function takes in sqlalchemy.MetaData or actions as inputs only." in str(
         e.value
     )
+
+
+postgres_async = create_postgres_fixture(async_=True)
+
+
+@pytest.mark.asyncio
+async def test_basic_postgres_fixture_async(postgres_async):
+    async for engine in postgres_async:
```

Comment:
what's up with this ^? I'm wondering if it's because of the `yield from`s?

I would expect this to "just work" as a session directly, unless this is how async sessions are supposed to work for some reason?

Reply:
This is an `async` problem. By using `yield` in `manage_async` we are producing an `async_generator`. This is propagated forward to the fixture all the way tests. The nature of async generators does not allow us to just consume it directly like synchronous ones.

Reply:
I understand that that's how it would work with no intervention, but i would expect pytest-asyncio to transparently handle this and make it work like vanilla fixtures.

The expected behavior of the plugin in the context of an async fixture with a yield point is underdocumented in their docs, but based on some grepping through their (issues)]https://github.com/pytest-dev/pytest-asyncio/issues/48) and [tests](https://github.com/pytest-dev/pytest-asyncio/blob/master/tests/async_fixtures/test_async_gen_fixtures.py), it seems like that's how it **should** work

Reply:
Based on [this](https://github.com/pytest-dev/pytest-asyncio/blob/master/tests/async_fixtures/test_async_gen_fixtures.py), you'll notice the `mock` fixture does not yield but return a value. Our `manage_async` is yielding values instead of returning values. Maybe this is an issue with the nested `async_generators`. I played around with `manage_async` by replacing the yield statements with return statements. It worked out just like the example you pointed out, however, I faced an issue with an unclosed session.

Reply:
I was looking specifically at `async_gen_fixture`, which is both `async` and `yields`. If you imagine it just `yield`ed some static value, i would imagine it'd be analagous to our `_async` fixture, since we have no nested equivalent fixture to `mock`

Reply:
I think it's because the fixture is yielding an async_generator. It's not consuming the value from `manage_async` then yielding again. It's just yielding the value from `manage_async` directly. I can add logic for consuming in the `_async` method so that end users don't have to deal with the for loop syntax.